src/app/code-panel/code-upload-panel.component.ts

```diff
@@ -20,13 +20,17 @@ export class CodeUploadPanelComponent implements OnInit {
 
   @ViewChildren('fileInput') private fileUploads: FileUpload[];
 
+  submissionService: SubmissionService;
 
   constructor(public studentService: StudentService,
               private problemService: ProblemService,
-              private submissionService: SubmissionService,
+              // private submissionService: SubmissionService,
               private route: ActivatedRoute,
               private messageService: MessageService,
+              private injector: Injector,
               private router: Router) {
+    const submissionServiceToken = route.snapshot.data['submissionService'];;
```

Comment:
I think `token` is misleading (refer to authentication)

rename `submissionServiceToken` to `submissionServiceInstanceName`